login.php

```diff
@@ -23,20 +35,23 @@
 
         if (password_verify($password, $db_password)) {
             $correct_password = true;
-            $_SESSION['user_id'] = $result['id_user'];
+            $_SESSION['id_user'] = $result['id_user'];
```

Comment:
what's the reason of having all user fields in a session? Wouldn't be the ID enough and then you can get the details?
What if the user has 20 fields, are we going to use all of them?

If we need these in the session, we might have an array, like this https://stackoverflow.com/questions/21652702/multidimensional-array-in-php-session

Reply:
There is not much data for this project. That's why I decided to keep them. 
My idea is not to re-query the database for query optimization.

I can redo the array as follows:  $_SESSION['user_data']['user_id'] 

When logging in, I can take only the ID and when I need the other data, I can send a new request to the database.

Which of the two options should I implement?

Reply:
let's say that Session is not quite reliable information and in 99% of the cases, it's better to call the DB. Not to mention that you can get the user data in one DB query and use it on all pages/files.

Let's update that, as it's way better, trust me

Reply:
I removed the variables from the session that are not sure when and if they will be used at all